autogluon/task/text_prediction/models/basic_v1.py

```diff
@@ -0,0 +1,911 @@
+    def default_config():
+        """Get the default configuration
+
+        Returns
+        -------
+        cfg
+            The configuration specified by the key
+        """
+        return base_cfg()
+
+    def train(self, train_data, tuning_data, resource,
+              time_limits=None,
+              scheduler='fifo',
+              searcher=None,
+              num_trials=10,
+              grace_period=None,
+              max_t=None,
+              reduction_factor=4,
+              brackets=1,
+              plot_results=False,
+              console_log=True,
+              ignore_warning=True):
+        start_tick = time.time()
+        logging_config(folder=self._output_directory, name='main',
+                       console=console_log,
+                       logger=self._logger)
+        assert len(self._label_columns) == 1
+        # TODO(sxjscience) Try to support S3
+        os.makedirs(self._output_directory, exist_ok=True)
+        search_space_reg = args(search_space=space.Dict(**self.search_space))
+        if scheduler == 'hyperband' and time_limits is None:
+            time_limits = 5 * 60 * 60  # 5 hour
+        train_fn = search_space_reg(functools.partial(train_function,
+                                                      train_data=train_data,
+                                                      time_limits=time_limits,
+                                                      tuning_data=tuning_data,
+                                                      base_config=self.base_config,
+                                                      problem_types=self.problem_types,
+                                                      column_properties=self._column_properties,
+                                                      label_columns=self._label_columns,
+                                                      label_shapes=self._label_shapes,
+                                                      log_metrics=self._log_metrics,
+                                                      stopping_metric=self._stopping_metric,
+                                                      console_log=console_log,
+                                                      ignore_warning=ignore_warning))
+        if scheduler == 'fifo':
+            if searcher is None:
+                searcher = 'random'
+            scheduler = FIFOScheduler(train_fn,
+                                      time_out=time_limits,
+                                      num_trials=num_trials,
+                                      resource=resource,
+                                      searcher=searcher,
+                                      checkpoint=None,
+                                      reward_attr='reward',
+                                      time_attr='time_spent')
+        elif scheduler == 'hyperband':
+            if searcher is None:
+                searcher = 'random'
+            if grace_period is None:
```

Comment:
Setting these things here feels like its a lot of duplicated code waiting to happen. Why not either have the defaults set in the actual FIFOScheduler/HyperbandScheduler init themselves, or have them set earlier on before it gets to this point. Are we planning every model to do this same logic and have different max_t/grace_period settings?

Reply:
I think the best "max_t" and "grace_period" won't be the same for different models so I manually picked a value here.